### PR TITLE
Add `--parallelism` parameter to `terraform apply`

### PR DESCRIPTION
**Pull request by nodeg**
This will

- add the `-parallelism` parameter to the execution of terraform.

```bash
$ terraform apply --help | grep parallelism -A1
  -parallelism=n         Limit the number of parallel resource operations.
                         Defaults to 10.
```

- Unit test added. Thank you @SchoolGuy for your help!
- Directory structure updated
- setup.py and requirements.txt added

### Links
- https://github.com/SUSE/spacewalk/issues/18101
- https://github.com/SUSE/susemanager-ci/pull/607

### CODE REVIEW

**Review by SchoolGuy — APPROVED** [review]
LGTM in my eyes. The fact that we don't document the method parameters in a machine readable way is a bit sad but let's not make that the topic of this PR.

**Comment by nodeg** [comment]
Yes. Better docstrings would make sense but I do not want to break the current style of the comments. I will create an issue for e.g. using Sphinx. Thank you for the review @SchoolGuy!

**Review by jordimassaguerpla — APPROVED** [review]
LGTM

**Review by juliogonzalez — CHANGES_REQUESTED** [review]
LGTM, but since now apply supports parameters, please add a unit test at `test/test_terraformer.py`

**Review by Bischoff — APPROVED** [review]

**Review by srbarrios — APPROVED** [review]

**Comment by nodeg** [comment]
- Unit test added. Thank you @SchoolGuy for your help!
- Directory structure updated
- `setup.py` and `requirements.txt` added

My unit test is successful although I see another test that fails. See https://github.com/uyuni-project/terracumber/issues/15
```bash
$ python3 -m unittest
......VAR=555
..Removing tags and fetching from <MagicMock name='Repository().remotes.__getitem__().url' id='140177728791184'>...
mytag seems to be a tag
Checking out
Removing tags and fetching from <MagicMock name='Repository().remotes.__getitem__().url' id='140177728791184'>...
Checking out
Performing hard reset to ignore local changes
.Removing tags and fetching from <MagicMock name='Repository().remotes.__getitem__().url' id='140177728787056'>...
Checking out
Performing hard reset to ignore local changes
....mytag seems to be a tag
mytag seems to be a tag
Checking out
Checking out
Performing hard reset to ignore local changes
.Removing tags and fetching from <MagicMock name='Repository().remotes.__getitem__().url' id='140177717667504'>...
mytag seems to be a tag
Checking out
...F.........
======================================================================
FAIL: test_sort_test_files_by_mtime (test.test_junit.TestJunit)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dgedon/git/terracumber/test/test_junit.py", line 19, in test_sort_test_files_by_mtime
    self.assertListEqual(self.junit.sort_test_files_by_mtime(), file_list)
AssertionError: Lists differ: ['tes[46 chars]dary-srv_delete_channel_from_ui.xml', 'test/re[485 chars]xml'] != ['tes[46 chars]dary-trad_centos_client.xml', 'test/resources/[485 chars]xml']

First differing element 0:
'test[18 chars]ailures/TEST-features-secondary-srv_delete_channel_from_ui.xml'
'test[18 chars]ailures/TEST-features-secondary-trad_centos_client.xml'

+ ['test/resources/junit/failures/TEST-features-secondary-trad_centos_client.xml',
+  'test/resources/junit/failures/TEST-features-secondary-srv_virtual_host_manager.xml',
- ['test/resources/junit/failures/TEST-features-secondary-srv_delete_channel_from_ui.xml',
? ^                                                           ^ -------------- ^^^^^

+  'test/resources/junit/failures/TEST-features-secondary-srv_users.xml',
? ^                                                           ^^  ^

+  'test/resources/junit/failures/TEST-features-secondary-srv_test_maintenance_windows.xml',
+  'test/resources/junit/failures/TEST-features-secondary-srv_power_management.xml',
   'test/resources/junit/failures/TEST-features-secondary-srv_delete_channel_with_tool.xml',
-  'test/resources/junit/failures/TEST-features-secondary-srv_power_management.xml',
?                                                             ^^^ ^ ^  ^^  ^^^     ^

+  'test/resources/junit/failures/TEST-features-secondary-srv_delete_channel_from_ui.xml']
?                                                             ^ ^^^^ ^^  ^ +++++ ^^^     ^

-  'test/resources/junit/failures/TEST-features-secondary-srv_test_maintenance_windows.xml',
-  'test/resources/junit/failures/TEST-features-secondary-srv_users.xml',
-  'test/resources/junit/failures/TEST-features-secondary-srv_virtual_host_manager.xml',
-  'test/resources/junit/failures/TEST-features-secondary-trad_centos_client.xml']

----------------------------------------------------------------------
Ran 27 tests in 0.096s

FAILED (failures=1)

```

**Review by SchoolGuy — APPROVED** [review]
I renew my faith in this PR. The new project structure is a lot better imho.

**Comment by nodeg** [comment]
@juliogonzalez Unit tests are implemented. Another review from your side would be appreciated.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from distutils.core import setup
```
Comment (juliogonzalez) [inline setup.py:1]: Small request: should we maybe document at `README.md` how to use this? :-)

Yeah, even if it's maybe quite obvious.

**Review by juliogonzalez — APPROVED** [review]
LGTM. I agree: the new structure looks waaaaaay better.

Feel free ignore handle my suggestion before merging.

File: setup.py (hunk 1 above)
Comment (nodeg) [reply setup.py:1]: I would add an issue for improving the README if that is ok 😉.

Comment (nodeg) [reply setup.py:1]: https://github.com/uyuni-project/terracumber/issues/16

**Pull request merged**